### PR TITLE
AMI resolver - owner array

### PR DESCRIPTION
**Pull request by herebebogans**
Resolver can take an owner array.
Not sure I have the spec exception test right.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -288,9 +288,12 @@ ${hostedzone://example.com/private}
 
 Looks up an AMI given a name prefix which may contain wildcards. If query returns multiple images the latest is used.
 
+Owners takes a query string list of AWS account ID, self (owner is the sender of the request), or an AWS owner alias (valid values are amazon | aws-marketplace | microsoft). Omitting this option returns all images for which you have launch permissions, regardless of ownership.
```
Comment (danielbergamin) [inline README.md]: Big fan of this, needed the flexibility to resolve images that are owned by another account and shared.

Comment (ampedandwired) [inline README.md]: The documentation here looks like it doesn't match the code. If owner is omitted, doesn't it default to "self"?

I think defaulting to "self" is probably the behaviour we want, as it's backwards compatible and less likelihood of ending up with the wrong image.

Comment (herebebogans) [inline README.md]: I copied that description from here .. https://docs.aws.amazon.com/sdkforruby/api/Aws/EC2/Client.html#describe_images-instance_method for options: owners

Comment (herebebogans) [inline README.md]: Ahh you are quite right ... I'll update the Readme in the PR.

**Comment by herebebogans** [comment]
Updated

**Comment by ampedandwired** [comment]
Looks good, thanks!

**Pull request merged**